contributions/whiten.py

```diff
@@ -0,0 +1,191 @@
+        return np.stack(white_segments, axis=0)
+
+    else:
+        raise ValueError("Strain numpy array dimension must be 1 or 2.")
+
+
+def main():
+    parser = ArgumentParser(description="Strain whitening script to be used for development of submission to the MLGWSC-1 mock data challenge. Written by Ondřej Zelenka.")
+
+    parser.add_argument('--verbose', action='store_true', help="Print update messages.")
+    parser.add_argument('--debug', action='store_true', help="Show debug messages.")
+
+    parser.add_argument('inputfile', type=str, help="The path to the input data file. It should be of the format generated by the generate_data.py script.")
+    parser.add_argument('outputfile', type=str, help="The path where to store the whitened data. The file must not exist.")
+    parser.add_argument('-s', '--segment-duration', type=float, default=0.5, help="Duration in seconds to use for each sample of the spectrum for PSD estimation. Default: 0.5.")
+    parser.add_argument('-d', '--max-filter-duration', type=float, default=0.25, help="Maximum length of the time-domain filter in seconds. Default: 0.25.")
+    parser.add_argument('--hard-trunc', action='store_true', help="Use a hard truncation of the time-domain filter. Otherwise, a Hann window is used.")
+    parser.add_argument('--keep-corrupted', action='store_true', help="Keep the region of the time series corrupted by the whitening. Otherwise, it is excised.")
+    parser.add_argument('-f', '--low-frequency-cutoff', type=float, default=20., help="Low frequency cutoff to be passed to the inverse spectrum truncation. This should be matched to a known low frequency cutoff of the data if there is one. Default: 20.")
+    parser.add_argument('--dtype', type=int, default=None, help="Data type specified by number of bits. If supplied, the input data will be cast to the corresponding floating point data type before whitening.")
+
+    args = parser.parse_args()
+
+    ### Set up logging
+    if args.debug:
+        log_level = logging.DEBUG
+    elif args.verbose:
+        log_level = logging.INFO
+    else:
+        log_level = logging.WARN
+    logging.basicConfig(format='%(levelname)s | %(asctime)s: %(message)s', level=log_level, datefmt='%d-%m-%Y %H:%M:%S')
+
+    ### Check existence of output file
+    if os.path.isfile(args.outputfile):
+        raise RuntimeError("Output file exists.")
+    else:
+        pass
+
+    if args.hard_trunc:
+        trunc_method = None
+    else:
+        trunc_method = 'hann'
+
+    logging.debug(("Initializing caster to data type %s" % str(args.dtype)))
+    caster = SafeCaster(args.dtype)
+
+    with h5py.File(args.inputfile, 'r') as infile, h5py.File(args.outputfile, 'x') as outfile:
+        copy_attrs(infile, outfile)
+        iterable1 = infile.items()
+        if args.verbose:
+            iterable1 = tqdm(iterable1, desc='Iterating over detectors')
+        for detector_group_name, in_detector_group in iterable1:
```

Comment:
Might be good to directly use tqdm here
```suggestion
        for detector_group_name, in_detector_group in tqdm(infile.items(), desc='Iterating over detectors', disabled=not args.verbose):
```